layers.go

```diff
@@ -1329,6 +1342,17 @@ func (r *layerStore) PutAdditionalLayer(id string, parentLayer *Layer, names []s
 	return copyLayer(layer), nil
 }
 
+func pickStoreLocation(jsonPath *[numLayerLocationIndex]string, volatile, writeable bool) layerLocations {
+	switch {
+	case volatile:
+		return volatileLayerLocation
+	case !writeable && jsonPath[indexFromLayerLocation(imageStoreLayerLocation)] != "":
```

Comment:
I think a previous version tried to place the ID-mapping layer to the same location as a parent … and, second thought, without more direct coordination with the overlay driver (see other parts of this review), that would be out of sync with the overlay driver? Fine.

Reply:
yes, we kind of use `writeable` as `container layer`.  Not sure if it makes sense to specify differently, but would be nice to move this logic out of the graph driver driver somehow so that it gets an explicit location where the layer must be created